### PR TITLE
Update readme to better explain System Registry

### PR DESCRIPTION
**Pull request by mobileoverlord**
WIP. Please proof read.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,73 +14,29 @@ def deps do
...
+The tree is navigated using a list of keys representing the path to the desired leaf node called a `scope`.
 
+Example:
 ```elixir
-{:ok, {%{a: 1}, %{}}} = SystemRegistry.update([:a], 1)
+%{state: %{network_interface: %{"wlan0" => %{ipv4_address: "192.168.1.100"}}}}
 ```
 
-Calls to `update/2` return a delta-state as a 2-tuple of `{new, old}`. Updates will either create keys (tree nodes) or replace their value.
+In this example, there is only a one leaf node, `ipv4_address`, located at the scope `[:state, :network_interface, "wlan0", :ipv4_address]`
```
Comment (tmecklem) [inline README.md]: Minor grammar: "only a one" probably meant to be "only one"

File: README.md [hunk 2]
```diff
@@ -14,73 +14,29 @@ def deps do
...
 
-```elixir
-{:ok, {%{a: 1}, %{}}} = SystemRegistry.update([:a], 1)
-{:ok, {%{a: 2}, %{a: 1}}} = SystemRegistry.update([:a], 2)
-```
+Processes contribute data to SystemRegistry by applying a transaction. A transaction can modify data by composing one or many calls to update, delete, or move. Registrants are notified of a change once the entire transaction has been successfully applied.
 
-If we instead want to have sub-keys `:b` and `:c` under the top-level `:a` key, we could do so like this:
+Data flows through SystemRegistry in two phases. First, process data is stored in a seperate fragment labeled by the caller pid and only contains the applied transactions of the caller. Second, the local pid fragment is then applied to the global state through a `SystemRegistry.Processor`.
```
Comment (tmecklem) [inline README.md]: Spelling: "seperate" should be "separate"

File: README.md [hunk 3]
```diff
@@ -14,73 +14,29 @@ def deps do
...
-  SystemRegistry.transaction
-  |> SystemRegistry.update([:a], 1)
-  |> SystemRegistry.update([:b], 2)
-  |> SystemRegistry.commit
-```
-
-#### Processors
-
-Processors are workers that can perform operations based on transactions. SystemRegistry ships with a default `State` processor. Processors implement the `Processor` behaviour and have the capability of being consulted during transaction validation and notified of committed transactions.
+Processors are workers that implement the `SystemRegistry.Processor` behaviour and are the only means of moving data from local fragments to the global state. Processors implement two callbacks methods: `handle_validate/2` and `handle_commit/2`. A transaction can only be committed if all processors return `:ok` during the validation sequence. Processor validation errors are accumulated and returned in the case of an unsuccessful commit. SystemRegistry automaticlaly starts two processors for state and config.
```
Comment (tmecklem) [inline README.md]: * "implement two callbacks methods" - callbacks should probably be "callback"
* "automatically" later in the paragraph

File: README.md [hunk 4]
```diff
@@ -141,31 +97,101 @@ config :system_registry, SystemRegistry.Processor.Config,
...
+
+SystemRegistry operates on a tree of nodes represented as nested maps, so if the value assigned to a scope is a Map, it is recursively expanded into scopes.
+
+```elixir
+{:ok, {%{a: %{b: 1}}, %{}}} = SystemRegistry.update([:a], %{b: 1})
+```
+
+**move**
+
+Nodes can be moved from one scopt to another. You can move both leaf nodes or internal nodes.
```
Comment (tmecklem) [inline README.md]: "scope" is misspelled

**Comment by mobileoverlord** [comment]
Thanks @tmecklem for the proof. Looks like I need a spellcheck plugin for my markdown :)

**Pull request merged**